### PR TITLE
feat: added debug flag in the SuperTokenConfig in the init() for logging

### PR DESCRIPTION
**Pull request by IamMayankThakur**
## Summary of change

Added debug flag in the SuperTokenConfig in the init() for logging

## Related issues

-   Link to issue1 here
-   Link to issue1 here

## Test Plan

Added unit tests.

## Documentation changes

(If relevant, please create a PR in our [docs repo](https://github.com/supertokens/docs), or create a checklist here highlighting the necessary changes)

## Checklist for important updates

-   [ ] Changelog has been updated
-   [ ] `coreDriverInterfaceSupported.json` file has been updated (if needed)
    -   Along with the associated array in `supertokens_python/constants.py`
-   [ ] `frontendDriverInterfaceSupported.json` file has been updated (if needed)
-   [ ] Changes to the version if needed
    -   In `setup.py`
    -   In `supertokens_python/constants.py`
-   [ ] Had installed and ran the pre-commit hook
-   [ ] Issue this PR against the latest non released version branch.
    -   To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    -   If no such branch exists, then create one from the latest released branch.
-   [ ] If have added a new web framework, update the `supertokens_python/utils.py` file to include that in the `FRAMEWORKS` variable
-   [ ] If added a new recipe that has a User type with extra info, then be sure to change the User type in supertokens_python/types.py
-   [ ] Make sure that `syncio` / `asyncio` functions are consistent.
-   [ ] If access token structure has changed
    -   Modified test in `tests/sessions/test_access_token_version.py` to account for any new claims that are optional or omitted by the core

### CODE REVIEW

File: supertokens_python/supertokens.py [hunk 1]
```diff
@@ -165,6 +170,9 @@ def __init__(
             mode,
         )
         self.supertokens_config = supertokens_config
+        self.debug = debug
```
Comment (rishabhpoddar) [inline supertokens_python/supertokens.py:173]: why do you need this variable?

File: supertokens_python/logger.py [hunk 2]
```diff
@@ -25,11 +24,16 @@
 
 supertokens_dir = path.dirname(__file__)
 
+
+def enable_debug_logging():
+    _logger.setLevel(logging.DEBUG)
```
Comment (rishabhpoddar) [inline supertokens_python/logger.py:29]: where are we resetting this during each of the test in this PR?

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: supertokens_python/logger.py (hunk 2 above)
Comment (IamMayankThakur) [reply supertokens_python/logger.py:29]: Added the code to set the logger to a higher level in the setup function, effectively disabling debug logs.
Also ordered the tests in a way which ensures the logger is reset every time.

File: supertokens_python/supertokens.py (hunk 1 above)
Comment (IamMayankThakur) [reply supertokens_python/supertokens.py]: It was unused, removed it. Thanks.

**Pull request merged**